### ISSUE TITLE
E117: Unknown function: TSOnBufEnter

### ISSUE THREAD

**Issue by adc17** [mhartington/nvim-typescript#139]
# Problems summary

Whenever I open a `.ts` file, I see the following error and the plugin doesn't work:

```
Error detected while processing BufEnter Auto commands for "*.ts":
E117: Unknown function: TSOnBufEnter
```

## Expected

Plugin to function normally.

## Environment Information

 * terminal: terminal:
iterm 3.1.5, tmux 2.6, zsh 5.4.2_1

 * vim/nvim version:
```
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
```

## Provide a minimal vim rc with less than 50 lines (Required!)

```vim
set nocompatible
filetype plugin indent on
call plug#begin('~/.local/share/nvim/plugged')
Plug 'leafgarland/typescript-vim', { 'for': 'typescript' }
Plug 'mhartington/nvim-typescript', {'for': ['typescript', 'tsx'], 'do': './install.sh' }
call plug#end()
```

## CheckHealth
```
health#nvim_typescript#check
========================================================================
## Find Node
  - OK: node is in $PATH

## Find Local Typescript
  - WARNING: No local server found, using global
    - ADVICE:
      - Install typescript locally for more accurate tooling
          $ npm install typescript --save-dev

## Find Global Typescript
  - OK: global tsserver is found

## Check for node bindings
  - OK: node bindings found

health#nvim#check
========================================================================
## Configuration
  - OK: no issues found

## Performance
  - OK: Build type: Release

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $XTERM_VERSION=''
  - INFO: $VTE_VERSION=''
  - INFO: $SSH_TTY=''

## tmux
  - OK: escape-time: 0ms
  - INFO: $TERM: screen-256color

health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: pbcopy

## Python 2 provider (optional)
  - INFO: `g:python_host_prog` is not set.  Searching for python2 in the environment.
  - INFO: Executable: /usr/local/bin/python2
  - INFO: Python2 version: 2.7.15
  - INFO: python2-neovim version: 0.2.6
  - OK: Latest python2-neovim is installed: 0.2.6

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Executable: /usr/local/bin/python3
  - INFO: Python3 version: 3.6.5
  - INFO: python3-neovim version: 0.2.6
  - OK: Latest python3-neovim is installed: 0.2.6

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.4.2p198 (2017-09-14 revision 59899) [x86_64-darwin17]
  - INFO: Host: /Users/me/.rbenv/shims/neovim-ruby-host
  - OK: Latest "neovim" gem is installed: 0.7.1

## Node provider (optional)
  - INFO: Node: v8.11.1
  - INFO: Host: /Users/me/n/bin/neovim-node-host
  - OK: Latest "neovim" npm is installed: 4.1.0
```

## The reproduce ways from neovim starting

Open any `.ts` file.

## Other notes

Initially I suspected the problem was `leafgarland/typescript-vim`, but on my second machine, I am running both of the applications together without any trouble.

**Comment by pyrho** [comment]
Did you run `:UpdateRemotePlugins` ?
Despite the health check reporting that it is up to date.

**Comment by adc17** [comment]
Yes, I tried that too.

**Comment by davidxmoody** [comment]
I have also been experiencing the same problem after updating my plugins yesterday. I've also tried `:UpdateRemotePlugins` but it doesn't seem to help.

**Comment by mhartington** [comment]
Not sure if there is any issue with caching from vim-plug, but a quick sanity check out be to remove the plugin from the installed location, and install it again.

Also, could you post your `:version` output? 

```
:version
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_H
AS_VAR_FROM -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wimplicit-fallthrough -Wvla -fstack-protect
or-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/build/config -I/tmp/neovim-20180209-92407-u
dkzoo/neovim-0.2.2/src -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/loca
l/opt/gettext/include -I/usr/include -I/tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/build/src/nvim/auto -I/tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/build
/include
Compiled by brew@HighSierra.local

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.2.2_1/share/nvim"

Run :checkhealth for more info
Press ENTER or type command to continue
```

**Comment by davidxmoody** [comment]
I just tried removing the plugin dir and reinstalling. I've got commit 11074c8 which appears to be the most recent one.

My `:version` output is:

```
:version
NVIM v0.2.2
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/local/Homebrew/Library/Homebrew/shims/super/clang -Wconversion -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_MSGPACK_HA
S_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gn
u99 -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -I/tmp/neovim-201802
09-92407-udkzoo/neovim-0.2.2/build/config -I/tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/src -I/usr/local/include -I/usr/local/include
 -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/include -I/usr/local/opt/gettext/include -I/usr/include -I/
tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/build/src/nvim/auto -I/tmp/neovim-20180209-92407-udkzoo/neovim-0.2.2/build/include
Compiled by brew@HighSierra.local

Features: +acl +iconv +jemalloc +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/local/Cellar/neovim/0.2.2_1/share/nvim"

Run :checkhealth for more info
```

**Comment by adc17** [comment]
@mhartington I did try wiping the directory but no luck. I may be running an older version of the plugin on the machine without problems, I'll check the two commit ids this evening (and post the `:version` output from the problem machine).

**Comment by adc17** [comment]
Sorry, this was a simple error on my side. The plugin was only activating when I entered a `.ts` file, and I was running `:UpdateRemotePlugins` from my vim config.

**Comment by NickHolt-Asana** [comment]
Could you describe how you fixed this? It sounds like you got this to work by opening a `.ts` file, and then running `:UpdateRemotePlugins`. This doesn't solve the issue for me, however.

**Comment by AntouanK** [comment]
same as @NickHolt-Asana 
I ran `: UpdateRemotePlugins`, I make sure everything is ok in `:checkhealth`, but the message keeps coming up whenever I open a .ts file in neovim.

**Comment by NickHolt-Asana** [comment]
I managed to fix this issue by running `install.sh` manually from the plugin folder, restarting Vim, and then running `:UpdateRemotePlugins`.

**Comment by niklaas** [comment]
Maybe the plugin should be configured with

```
Plug 'mhartington/nvim-typescript', {'do': 'sh install.sh'}
```
to run `install.sh` automatically when installing the plugin.

**Comment by acro5piano** [comment]
Thanks! I had the same problem but fixed thanks to you all.

**Comment by cmcaine** [comment]
This line tells vim-plug to do both for you:

```viml
Plug 'mhartington/nvim-typescript', {'do': ':!install.sh \| UpdateRemotePlugins'}
```

**Comment by kinowarrior** [comment]
Just adding in case useful. My problem was that I use virtual envs for Python, and hadn't activated the correct env. Opening a `.ts` file and running `:UpdateRemotePlugins` (once the python env was activated in the same shell) resolved the issue.

**Comment by MarianVasile** [comment]
In my case, i'm using nvm to change the node version between projects. As nvim/vim uses the system node (macos), you'll need to run:

`cd /Users/mark/.cache/vimfiles/repos/github.com/mhartington/nvim-typescript`
`nvm use system`
`install.sh`

And the same goes for the neovim package:

`nvm use system`
`npm i -g neovim`

I hope this helps :)
